### PR TITLE
Use pytest directly without tox

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #4055, #2292

#### What's this PR do?
This PR removes tox and runs pytest directly.

#### How should this be manually tested?
1. Run `docker compose exec web pytest` locally. It should pass.

#### Any background context you want to provide?
The local testing environment is now a bit less clean / less consistent between developers: Running tests locally will use your local `web` container environment variables, plus some overrides in `pytest.ini`. IMO, this is fine (and is what we do in other projects). The CI env is still very consistent.

### CODE REVIEW

File: scripts/test/test_suite.sh [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env bash
 set -euf -o pipefail
 
-docker-compose run web tox
-docker-compose run watch npm run-script lint
-docker-compose run watch npm run-script flow
-docker-compose run watch npm run-script coverage
```
Comment (ChristopherChudzicki) [inline scripts/test/test_suite.sh:7]: There is not currently a coverage command.

File: scripts/test/python_tests.sh
Comment (ChristopherChudzicki) [inline scripts/test/python_tests.sh:1]: This is what we use in other non-tox projects, e.g., https://github.com/mitodl/ocw-studio/blob/master/scripts/test/python_tests.sh

_`black` runs separately on CI, so I didn't include it in this._

File: tox.ini [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-    OPEN_DISCUSSIONS_DEFAULT_SITE_KEY=mm_test
-    OPEN_DISCUSSIONS_SECURE_SSL_REDIRECT=False
-    OPEN_DISCUSSIONS_NOTIFICATION_EMAIL_BACKEND=anymail.backends.test.EmailBackend
-    OPEN_DISCUSSIONS_BASE_URL=http://localhost:8063/
-    MAILGUN_SENDER_DOMAIN=other.fake.site
-    MAILGUN_KEY=fake_mailgun_key
-    OPENSEARCH_URL=http://localhost:9300/
-    OPENSEARCH_INDEX=testindex
-    INDEXING_API_USERNAME=mitodl
-    OCW_WEBHOOK_KEY=fake_key
```
Comment (mbertrand) [inline tox.ini:41]: Some of these aren't in the ci.yml file and/or pytest.ini file, might be good to make them all consistent?  For example `OPEN_DISCUSSIONS_NOTIFICATION_EMAIL_BACKEND` (xpro, mitxonline have the equivalent env in both ci.yml and pytest.ini)

**Review by mbertrand — CHANGES_REQUESTED** [review]
Works great, just a couple suggestions, but since tests are passing on CI anyway they probably aren't critical

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -68,14 +68,11 @@ jobs:
...
         run: ./scripts/test/stub-data.sh
-
       - name: Tests
         run: |
           export MEDIA_ROOT="$(mktemp -d)"
-          tox
+          ./scripts/test/python_tests.sh
         env:
-          DEBUG: 'False'
-          NODE_ENV: 'production'
```
[the remark this answers is not in the document]
Comment (ChristopherChudzicki) [reply .github/workflows/ci.yml:78]: - `NODE_ENV` shouldn't affect the python tests at all, so I don't see any point in setting it.
- `DEBUG=false` was previously being set in both `tox.ini` and `ci.yaml`, with same value. Now it's set in `pytest.ini`. I don't think there's any reason to set it in both places.

**Comment by ChristopherChudzicki** [comment]
@mbertrand I did start a discussion post about the `pytest-dotenv` idea I mentioned in slack: https://github.com/mitodl/hq/discussions/1836

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**